application/src/main/resources/ecc.yml

```diff
@@ -32,8 +32,12 @@ connection:
     ## Connection is tried every five second until either the timeout time passes or the connection is successful.
     ##
     timeout:
-      time: 0
+      time: 60
       unit: seconds
+    retryPolicy:
+      maxAttempts: 5
+      delay: 5
+      maxDelay: 30
```

Comment:
Yes you can, but considering an overloaded cluster, as the delay is calculated based on the attempt count, it can return a a really big number depending on the configuration

Reply:
In the default case above, the practical max delay will be 25 (5*5), so 30 will never happen. If you set attempts/delay to something bigger than 5*5 - like 6*6 - you will already know that the max delay of 30 will be hit, so it is meaningless to set attempts/delay to anything bigger. And not to forget, you still have the overall timeout to consider.

I'm just trying to understand the actual benefit here for introducing the entire RetryPolicy class as it is. For instance, why don't you just use the current timeout setting/functionality as it is, and then "divide" that into dynamic delays instead of the hardcoded delay of 5 for each retry? So, you could do the first retry after 5 secs, the next after 10 secs and so on until the timeout has been spent.

Example, you could add like a retry_delta setting if you want the back off time to grow for each retry. So, a delta of 5 would give the scenario above (5, 10, 15, etc) and a setting of 0 would give the current behaviour (5, 5, 5, etc).

    timeout:
      time: 60
      retry_delta: 5
      unit: seconds

The DefaultNativeConnectionProvider constructor could probably be a bit refactored as well - in general- to be cleaner than it is now.